pages/_app.js

```diff
@@ -1,8 +1,49 @@
+// Import external resources
+import { SWRConfig } from "swr";
+import { useImmer } from "use-immer";
+
+// Import internal resources
 import GlobalStyle from "@/styles";
 import Head from "next/head";
-import { SWRConfig } from "swr";
+import Layout from "@/components/Layout";
 
 export default function App({ Component, pageProps }) {
+  const [cartItems, updateCartItems] = useImmer([]);
+
+  // Duplication is far cheaper than the wrong abstraction.
```

Comment:
I don't really understand what's going on here but it looks amazing.

Reply:
```javascript
const sumUpArray = (accumulator, currentValue) => accumulator + currentValue;
const sumOfAllItemAmount = cartItems
    .map((item) => item.amount)
    .reduce(sumUpArray, 0);
```

`sumOfAllItemAmount` : Name of a const 
`.map((item) => item.amount)` : takes an array of object and returns a new array only with amounts type number
`reduce(sumUpArray, 0)` : takes the new array takes a initial value `0` continues in the array and adds the currentValue to it, returns the sum of all number of the new array

```javascript
beforeMapping = [{amount : 1}, {amount: 2}, {amount: 3}]
afterMapping = [1,2,3]
afterReduce = 6
```